### ISSUE TITLE
feature request: clone a particle

### ISSUE THREAD

**Issue by manuel-84** [gantry/gantry5#2168]
for future releases it would be nice if it will be possible to clone existing particles in the layout editor (i'm using the grav version)

**Comment by manuel-84** [comment]
I know you can clone particles from other outlines but what I mean is to clone a particle from the current outline you are editing

**Comment by w00fz** [comment]
Agree with @manuel-84.
We do need this desperately. I didn’t realized how annoying it was not having this option until I had to demo gantry at a conference. We do need a way to be able to duplicate or at least allow cloning particles within the same outline.
The only sane way right now is to go to the yaml file and manually do so.

**Comment by marktaylor46** [comment]
Not wishing to hijack this issue - but another thing that comes up often in the forums related to this is when you want to move a particle to a joomla module (because you want to use Joomla ACL for example) - you can't you have to start all over again in a "Gantry 5 particle" module. It would be great if there were a way to select a particle and click on "make this a module" button.

**Comment by kat05** [comment]
+1 this would be a great addition.

Also requested here: http://www.rockettheme.com/forum/joomla-template-isotope/275430-duplicate-a-particle?start=0#1379562

**Comment by pau1a** [comment]
+1 on this feature.

**Comment by yellowwebmonkey** [comment]
+1 as well

**Comment by N8Solutions** [comment]
> @marktaylor46 
> 
> Not wishing to hijack this issue - but another thing that comes up often in the forums related to this is when you want to move a particle to a joomla module (because you want to use Joomla ACL for example) - you can't you have to start all over again in a "Gantry 5 particle" module. It would be great if there were a way to select a particle and click on "make this a module" button.

I needed this feature as well a couple of days ago.

+1 for the implementation

**Comment by lausianne** [comment]
Clone! Please! Has this request been noticed by any of the Gantry masters yet?

**Comment by dmleeman** [comment]
Cannot answer for Grav or Wordpress, but Joomla does effectively allow you to do this via the clone module function in the Joomla > Extensions > Modules admin.

As long as you create all of your positions as Module Positions and then in each Module Position, add your desired Particle, this is easy.

When you clone the module, it creates a perfect copy of the contents .

It just takes a little more forward planning, but that has never hurt any project!

It also offers the designed advantage of creating a Base Outline with all of your site's required Module Positions in place (complete with Particles) and then simply use the modules' own menu assignment to select which Menu Item generated page that they will display.

I've created two ecommerce stores this way, complete with blogs and galleries, sticky scrolling sidebars in just one layout. 

ideally, you would need to know a bit of yaml to achieve the base outline you need to cover all layouts in your site, but the result is astoundingly fast.

I suspect that this was the original idea (of designing Joomla based sites at least).

**Comment by wdburgdorf** [comment]
Thanks for your comment and your insights. What you describe is more or less the standard Joomla way, as I understand it. If that is the best way, then I don't see the point of the layout builder or particles in the first place. They don't simplify anything then. I would actually find it easier to type a bit of html and php, and have complete control like that. The idea for me of a page builder is that I essentially don't need to plan or touch code. Except for minor tweaks. Of course I use a good template to start with. But when I actually use a page builder, then because I want a simple quick solution for a low budget site with little custom coding. 
Sure, we can clone modules. But particles have lots of useful parameters that would lose their purpose if all we use particles for  is placing modules. Therefore I'd find it useful if particles could be cloned, too.

**Comment by dmleeman** [comment]
"But particles have lots of useful parameters that would lose their purpose if all we use particles for is placing modules."

You lose no functions of the Particles at all, and it is the other way around; you are using Modules to position Particles - along with all of the functionality that the Particle allows. By using the Gantry5 Module in Joomla, then applying the desired Particle within you get all of the functionality of The Joomla modules (position, page, start and end publishing, caching, class suffixes) plus all of the inner Particle's own abilities and functions.

"They don't simplify anything then. I would actually find it easier to type a bit of html and php, and have complete control like that"

I wouldn't presume to tell a developer skilled in PHP, and javascript how to build their sites, but it is worth pointing out these undocumented abilities to those who are getting started, or know some HTML and CSS, how they can improve the speed and flexibility of G5 in the Joomla environment, and it's advantages over other builders.

"The idea for me of a page builder is that I essentially don't need to plan or touch code. Except for minor tweaks" - precisely what the above approach achieves.

"I'd find it useful if particles could be cloned, too." - Agreed, but I was pointing out to those who may not be aware, this function is, effectively, present - in Joomla at least.

**Comment by wdburgdorf** [comment]
Thanks a lot, interesting idea, makes more sense now. Spontaneously, it sounds like adding a level of complexity. I need think about it some more and try ...

**Comment by dmleeman** [comment]
No problem, at first the idea does seem counterintuitive. So, and this is for those following the issue
In Joomla, plan your layout as if using the Particles

1) Create you layout using the Module Position Particle - name them however you prefer
2) Create and apply Joomla Gantry5 Modules in each of the positions
3) Add the required Particle to your Module and set up as required - you now have a 'master' Particle set up within a Joomla Module, so you have date control, caching, class suffix, Menu Assignment, Permissions and Chrome PLUS all of the parameters of the inner Particle.
4) Clone the module as needed and apply to the positions set up in your page layout. Particle settings are cloned along with the Module

This way you an create any site with One Layout

**Comment by marktaylor46** [comment]
There is a way of cloning a particle on the same outlie - it's a fudge and I'm not claiming that it is elegant but it does work. If you're interested here is how:

1. Create a new outline in outline manager - call it anything you like and DO NOT assign it to any menu item item or make it default. It's probably best to use the "default" layout preset,
2. Go to the layout of that new outline > layout and press the "clear" button top right (full layout clear). and press SAVE.
3. Add an instance of the same type of particle that you wish to clone anywhere on the layout of this outline. There's no need to fill in any of the settings - just go to the "inheritance" tab, pick the outline you want to clone the paticle from and pick "clone" and tick the "particle attributes" and "block attributes" fields and press SAVE.
4. Go to the original outline on which you wish to clone the particle, Add an instance of the same particle type to the place on the layout that you want it and again go to the inheritance tab, pick "clone" and tick the two attributes fields again but this time choose the new outline that you created.

So following the above you can clone a particle on the same layout (you just have to go through an intermediary step). I know it's not elegant but it works and it can save you a lot of time if it's a particularly complex set of content in the particle.

I'd also say that actually cloning particles on the same outline layout is pretty rare.

I hope that helps.

Regards, Mark.

**Comment by dmleeman** [comment]
"I'd also say that actually cloning particles on the same outline layout is pretty rare." - I agree, true. It seems to be larger sites/e-commerce sites that the problem arises. Or later additions to an existing site -then it is *very* useful.

**Comment by marktaylor46** [comment]
Yes, that's my finding too. Things like "productlist" particle in Versla template where you've but masses of effort into entering lots of information in the particle. But even then why would you want two "productlist" particles on the same page?

Anyway - there's a workaround for anyone who has a deperate need. :)
